enterprise/server/remote_execution/containers/podman/podman.go

```diff
@@ -119,6 +121,21 @@ additionallayerstores=["/var/lib/soci-store/store:ref"]
 		if err := disk.WaitUntilExists(context.Background(), sociStoreDir, disk.WaitOpts{}); err != nil {
 			return nil, status.UnavailableErrorf("soci-store failed to start: %s", err)
 		}
+
+		// TODO(iain): there's a concurrency bug in soci-store that causes it
+		// to die occasionally. Report the executor as unhealthy if the
+		// soci-store directory doesn't exist for any reason.
+		env.GetHealthChecker().AddHealthCheck(
+			"soci_store", interfaces.CheckerFunc(
+				func(ctx context.Context) error {
+					if _, err := os.Stat(sociStorePath); err == nil {
```

Comment:
Have you verified that os.Stat() returns an error? Just want to check that a fuse error leads to a Stat error because technically the mount point is still there.

Reply:
Yeah, I confirmed it returns a "transport not connected" error